src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,31 @@
+package duke.command;
+
+import duke.exception.DukeInvalidListNumberInputException;
+import duke.exception.DukeLoadingErrorException;
+import duke.messages.Output;
+import duke.storage.Storage;
+import duke.task.TaskList;
+
+/**
+ * Represents a command to delete a task.
+ */
+public class DeleteCommand extends Command {
+
+    /**
+     * Class constructor.
+     *
+     * @param input A string representing the user input.
+     */
+    public DeleteCommand(String input) {
+        super("delete", input);
+    }
+
+    @Override
+    public CommandResult execute(TaskList tasks, Output output, Storage storage)
+            throws DukeInvalidListNumberInputException, DukeLoadingErrorException {
```

Comment:
Would it be better if the DukeInvalidListNumberInputException was given a slightly shorter name, perhaps DukeInvalidIndexException?

Reply:
That makes sense!